### DISCUSSION TITLE
How to discretize the differential equations?

### DISCUSSION THREAD

**Discussion by nychiang** [CURENT/andes#276] [General]
Hi Dr. Cui,

I wonder how the differential equations is solved in Andes, by finite element?
Can we control the discretization?

Cheers,

NY

**Comment by cuihantao** [comment]
The differential equations in Andes are DAEs, not PDEs. Finite elements is for PDE.

The equations in ANDES is solved by simultaneous methods that discretion the DAE into a set of nonlinear algebraic equations. See Prof. Federico Milano’s book.

Regards,
Hantao Cui

**Comment by cuihantao** [comment]
Here's the implementation for Trapezoidal method: `calc_q` is to build the residual equation after discretization, and `calc_jac` is to calculate the corresponding Jacobian.

These two functions rely on other functions in `tds.py` to calculate the values for the numerical DAE.

https://github.com/cuihantao/andes/blob/b8b9212c0d2fa24f6af81e1fca3df1034470eea4/andes/routines/daeint.py#L207-L234

**Comment by nychiang** [comment]
Got it! Thanks a lot for your quick response.
Cheers!
